### PR TITLE
Remove obsolete developer settings.

### PR DESCRIPTION
**Pull request by capnfabs**
Remove "Update local observations on submission" and "Sync observations incrementally" settings.

These settings have been 'on' by default for a long time, and we're relying on the behaviour of both
of them extensively.

Also:
- rename some methods named `startIncrementalObsSync` to `startObservationsSync`, because all
  observations syncs are incremental syncs now.
- delete unused `OdkActivityLauncher#showSavedXform()` method.

### CODE REVIEW

File: app/src/main/res/xml/pref_developer.xml [hunk 1]
```diff
@@ -9,20 +9,6 @@
...
-        android:summary="@string/pref_desc_xform_update_client_cache"
-        android:defaultValue="@bool/xform_update_client_cache_default" />
-
-    <!-- Whether to instantly update the observations cache incrementally -->
-    <CheckBoxPreference
-        android:key="incremental_observation_update"
-        android:title="@string/pref_title_incremental_observation_update"
-        android:summary="@string/pref_desc_incremental_observation_update"
-        android:defaultValue="@bool/incremental_observation_update_default" />
-
```
Comment (llvasconcellos) [inline app/src/main/res/xml/pref_developer.xml:25]: Aren't you going to remove the strings as well?

**Comment by llvasconcellos** [comment]
Good to merge after removing unused strings.

File: app/src/main/res/xml/pref_developer.xml (hunk 1 above)
Comment (capnfabs) [inline app/src/main/res/xml/pref_developer.xml:25]: Good catch. Done.

**Comment by G1DR4** [comment]
good to go too

File: app/src/main/res/xml/pref_developer.xml (hunk 1 above)
Comment (llvasconcellos) [inline app/src/main/res/xml/pref_developer.xml:25]: :thumbsup:

**Pull request merged**